ENIGMAsystem/SHELL/Platforms/Cocoa/CocoaJoystick.cpp

```diff
@@ -0,0 +1,72 @@
+/** Copyright (C) 2016 Rodrigo Rocha
+***
+*** This file is a part of the ENIGMA Development Environment.
+***
+*** ENIGMA is free software: you can redistribute it and/or modify it under the
+*** terms of the GNU General Public License as published by the Free Software
+*** Foundation, version 3 of the license or any later version.
+***
+*** This application and its source code is distributed AS-IS, WITHOUT ANY
+*** WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+*** FOR A PARTICULAR PURPOSE. See the GNU General Public License for more
+*** details.
+***
+*** You should have received a copy of the GNU General Public License along
+*** with this code. If not, see <http://www.gnu.org/licenses/>
+**/
+
+#include <string>
+using namespace std;
+
+namespace enigma
+{
+  //stub
+}
+
+/* Note: empty stub functions to build Mac executables without errors (and no joystick support, of course) */
+namespace enigma_user
+{
+  bool joystick_load(int id) {
+    return false; //stub
```

Comment:
It's just sad that parts of the engine are forcibly compiled into your game that introduce dependencies on joysticks. It's a smell—this means that LTO cannot remove those functions from your executable. I dislike stubs because I'd prefer you get compile errors rather than have a game built that does nothing when you tell it to load a joystick. That's just frustrating.

Still, this problem has nothing to do with your pull request, so no action to take, here.